[Keycloak CI] - BruteForceTest.testPermanentLockout failures

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

login/ui

### Describe the bug

Reference run: https://github.com/keycloak/keycloak/actions/runs/10624544379/job/29453219058

```
Error: OR] org.keycloak.testsuite.forms.BruteForceTest.testPermanentLockout  Time elapsed: 9.977 s  <<< FAILURE!
-- java.lang.AssertionError: Event expected
-- 	at org.junit.Assert.fail(Assert.java:89)
-- 	at org.junit.Assert.assertTrue(Assert.java:42)
-- 	at org.junit.Assert.assertNotNull(Assert.java:713)
-- 	at org.keycloak.testsuite.AssertEvents.poll(AssertEvents.java:87)
-- 	at org.keycloak.testsuite.forms.BruteForceTest.testPermanentLockout(BruteForceTest.java:570)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
-- 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
-- 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
-- 	at org.jboss.arquillian.junit.Arquillian$8$1.invokeMethod(Arquillian.java:319)
-- 	at org.jboss.arquillian.junit.MethodInvoker$1.invoke(MethodInvoker.java:18)
-- 	at org.jboss.arquillian.container.test.impl.execution.LocalTestExecuter.execute(LocalTestExecuter.java:57)
-- 	at org.keycloak.testsuite.arquillian.ModelTestExecutor.execute(ModelTestExecutor.java:49)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
-- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
-- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
-- 	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
-- 	at org.jboss.arquillian.container.test.impl.execution.ClientTestExecuter.execute(ClientTestExecuter.java:50)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
-- 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createContext(ContainerEventController.java:128)
-- 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createTestContext(ContainerEventController.java:118)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:116)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
-- 	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.test(EventTestRunnerAdaptor.java:139)
-- 	at org.jboss.arquillian.junit.MethodInvoker.invoke(MethodInvoker.java:15)
-- 	at org.jboss.arquillian.junit.Arquillian$8.evaluate(Arquillian.java:326)
-- 	at org.jboss.arquillian.junit.Arquillian$4.evaluate(Arquillian.java:198)
-- 	at org.jboss.arquillian.junit.Arquillian.multiExecute(Arquillian.java:344)
-- 	at org.jboss.arquillian.junit.Arquillian.access$200(Arquillian.java:48)
-- 	at org.jboss.arquillian.junit.Arquillian$5.evaluate(Arquillian.java:209)
-- 	at org.keycloak.testsuite.AssertEvents$1.evaluate(AssertEvents.java:79)
-- 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
-- 	at org.jboss.arquillian.junit.Arquillian$7$1.invoke(Arquillian.java:273)
-- 	at org.jboss.arquillian.container.test.impl.execution.ClientBeforeAfterLifecycleEventExecuter.execute(ClientBeforeAfterLifecycleEventExecuter.java:88)
-- 	at org.jboss.arquillian.container.test.impl.execution.ClientBeforeAfterLifecycleEventExecuter.on(ClientBeforeAfterLifecycleEventExecuter.java:66)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
-- 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createContext(ContainerEventController.java:128)
-- 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createBeforeContext(ContainerEventController.java:114)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:116)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
-- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
-- 	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.fireCustomLifecycle(EventTestRunnerAdaptor.java:159)
-- 	at org.jboss.arquillian.junit.Arquillian$7.evaluate(Arquillian.java:267)
-- 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
-- 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
-- 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
-- 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
-- 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
-- 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
-- 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
-- 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
-- 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
-- 	at org.jboss.arquillian.junit.Arquillian$2.evaluate(Arquillian.java:160)
-- 	at org.jboss.arquillian.junit.Arquillian.multiExecute(Arquillian.java:344)
-- 	at org.jboss.arquillian.junit.Arquillian.access$200(Arquillian.java:48)
-- 	at org.jboss.arquillian.junit.Arquillian$3.evaluate(Arquillian.java:171)
-- 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
-- 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
-- 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
-- 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
-- 	at org.jboss.arquillian.junit.Arquillian.run(Arquillian.java:109)
-- 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:377)
-- 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:302)
-- 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:248)
-- 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:167)
-- 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
-- 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
-- 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
-- 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
```

### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

Succesful test

### Actual behavior

Failing test

### How to Reproduce?

Run tests multiple times / Run tests in GHA

### Anything else?

_No response_

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/BruteForceTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/BruteForceTest.java
@@ -558,6 +558,7 @@ public class BruteForceTest extends AbstractTestRealmKeycloakTest {
         try {
             // arrange
             realm.setPermanentLockout(true);
+            realm.setQuickLoginCheckMilliSeconds(0L);
             testRealm().update(realm);
 
             // act
@@ -567,7 +568,7 @@ public class BruteForceTest extends AbstractTestRealmKeycloakTest {
             // As of now, there are two events: USER_DISABLED_BY_PERMANENT_LOCKOUT and LOGIN_ERROR but Order is not
             // guarantee though since the brute force detector is running separately "in its own thread" named
             // "Brute Force Protector".
-            List<EventRepresentation> actualEvents = Arrays.asList(events.poll(), events.poll(), events.poll());
+            List<EventRepresentation> actualEvents = Arrays.asList(events.poll(), events.poll());
             assertIsContained(events.expect(EventType.USER_DISABLED_BY_PERMANENT_LOCKOUT).client((String) null).detail(Details.REASON, "brute_force_attack detected"), actualEvents);
             assertIsContained(events.expect(EventType.LOGIN_ERROR).error(Errors.INVALID_USER_CREDENTIALS), actualEvents);
 

